app/models/agents/post_agent.rb

```diff
@@ -162,22 +171,29 @@ def normalize_response_headers(headers)
       }
     end
 
-    def handle(data, payload = {}, headers)
-      url = interpolated(payload)[:post_url]
+    def handle(data, event = Event.new, headers)
+      url = interpolated(event.payload)[:post_url]
 
       case method
       when 'get', 'delete'
         params, body = data, nil
       when 'post', 'put', 'patch'
         params = nil
 
-        case (content_type = interpolated(payload)['content_type'])
+        content_type = if has_file_pointer?(event)
```

Comment:
Hah. I don't feel strongly, go with whichever you prefer.  Personally, I find it confusing that the `if` has both a side effect (assigning into `data`) and is also being used for assignment itself.  The new one is fine too.